python/mlcroissant/mlcroissant/_src/operation_graph/operations/download_test.py

```diff
@@ -134,15 +134,34 @@ def test_hashes_do_not_match():
 
 
 @pytest.mark.parametrize("conforms_to", CroissantVersion)
-def test_hashes_do_match(conforms_to):
+# Test the hex and base64 hash values
+@pytest.mark.parametrize("hash_value", ["e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855", "47DEQpj8HBSa+/TImW+5JCeuQeRkm5NMpJWZG3hSuFU="])
+def test_sha256_hashes_do_match(conforms_to, hash_value):
     with tempfile.NamedTemporaryFile(delete=False) as f:
         filepath = f.name
         metadata = Metadata(name="bar", conforms_to=conforms_to)
         file_object = create_test_file_object(
             name="foo",
             content_url=os.fspath(filepath),
             # Hash will match!
-            sha256="e3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855",
+            sha256=hash_value,
+        )
+        file_object.parents = [metadata]
+        download = Download(operations=operations(), node=file_object)
+        download()
+
+@pytest.mark.parametrize("conforms_to", CroissantVersion)
+# Test the hex and base64 hash values
+@pytest.mark.parametrize("hash_value", ["d41d8cd98f00b204e9800998ecf8427e", "1B2M2Y8AsgTpgAmY7PhCfg=="])
+def test_md5_hashes_do_match(conforms_to, hash_value):
```

Comment:
I tried a version of this with a parameter to control the `hash_algo`, but I didn't like how that read since it's just string matching against `"md5"` and `"sha256"` to assign a different `file_object = create_test_file_object`. No type safety there (unlike what you have for `CroissantVersion`) pushed me in the separate tests direction. I think this seems fine, though?

Reply:
create_test_file_object is a helper. If it's not handy in that case, you can just use:

```
        file_object = FileObject(
            ...,
            md5=hash_value,
            sha256=None,
        )

        file_object = FileObject(
            ...,
            md5=None,
            sha256=hash_value,
        )
```

You can also also use [mock](https://github.com/mlcommons/croissant/blob/c0b09bb8a045e71515f526bb8a23a6c6873a7bc8/python/mlcroissant/mlcroissant/_src/operation_graph/execute_test.py#L67) to mock hash_algo, although I think it's not as powerful as the test described above.

Reply:
It's less about the helper and more about not wanting to have to write code that looks like this:
```
file_object = None
if hash_algo == "md5":
  file_object = <build md5 FileObject without sha256>
elif hash_algo == "sha256":
  file_object = <build sha256 FileObject without md5>
```

Or with a `switch`/`case`, same difference. I usually see logical branches in the test setup code as a sign that a separate test should be broken out.